### PR TITLE
Loottables

### PR DESCRIPTION
**Pull request by inxomnyaa**
Chests 100% generated from loottables. Uses a MCPE default addon as submodule (poggit dependency update maybe? idk how poggit actually functions)

This also fixes where the chests are placed and their rotation ( in temples)
adding new variants is easy AF.

### CODE REVIEW

**Comment by Ad5001** [comment]
Can this be merged?

**Comment by inxomnyaa** [comment]
i talked to @dktapps and looks like there are legal issues.. we could ignore them for now and say, hey mojang, the addons are open source anyways. otherwise.. it could be made an external link instead.

**Comment by inxomnyaa** [comment]
So, i leave the decision to you ;) i'd say yes

**Comment by Ad5001** [comment]
They didn't specify any LICENSE. And the problem is that all rights are reserved to the author in case of no LICENSE. + The truth of dependencies is they broke on download. It'd be better just to take the loot tables I believe.

**Comment by Thunder33345** [comment]
just edit the odds by 1% shouldnt that stops it?

File: resources/loot_tables/abandoned_mineshaft.json [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+                "min": 1,
+                "max": 4
+              }
+            }
+          ],
+          "weight": 5
+        },
+        {
+          "type": "item",
+          "name": "minecraft:torch",
```
Comment (inxomnyaa) [inline resources/loot_tables/abandoned_mineshaft.json:261]: OMFG!!!! THATS EVEN WORSE! That makes US the owners of the files instead of a submodule!

**Review by inxomnyaa — COMMENTED** [review]
pllllllzzzz stoooooppppp

**Comment by Ad5001** [comment]
@thebigsmileXD Two problems with submodules:
- Not cloning when downloading
- Adds useless things such as entity generation

+ We can make a "special thanks" in the readme telling that we are not the copyright owners of content under /resources/loot_tables.

**Comment by Ad5001** [comment]
+ You forgot something with your debug. It's executed sync whereas generation is async so we cannot spawn tiles. That's my main struggle.

**Comment by inxomnyaa** [comment]
thats why the main class comes in. AS SAID, this pushed the Legal thing to a NOPE

**Comment by Ad5001** [comment]
You haven't understood what I meant. Your "placeLootTable" method uses a pocketmine\level\Level instance to spawn a tile. In async, there is no pocketmine\level\Level. Just ChunkManagers which cannot spawn tiles.

**Comment by Ad5001** [comment]
And about the legal thing, I have to agree with you. But the github bug makes it absolutly unuser friendly.

**Comment by inxomnyaa** [comment]
There is no github bugs, there are simply users that are too noobish to git clone recursive..

> You haven't understood what I meant. Your "placeLootTable" method uses a pocketmine\level\Level instance to spawn a tile. In async, there is no pocketmine\level\Level. Just ChunkManagers which cannot spawn tiles.

Wrong. it uses the ChunkManager, which _is_ a Level class

**Comment by Ad5001** [comment]
> There is not github bugs, there are simply users that are too noobish to git clone recursive..

Well, that means to download a simple plugin, you need to pass by git? Well, the common server owners *don't know how to use git*. They simply click the big "Download" button which is generated from an archive, which, doesn't apply submodules and transfer it to their server using a GUI such as Filezilla.
> Wrong. it uses the ChunkManager, which is a Level class

https://github.com/pmmp/PocketMine-MP/blob/master/src/pocketmine/level/Level.php#L112
https://github.com/pmmp/PocketMine-MP/blob/master/src/pocketmine/level/SimpleChunkManager.php#L28
Level *is* a ChunkManager thought ChunkManager (in case of generation SimpleChunkManager) *isn't* a Level.

**Comment by Ad5001** [comment]
Btw why did you deleted this branch?

File: resources/loot_tables/abandoned_mineshaft.json (hunk 1 above)
Comment (AvgZing) [reply resources/loot_tables/abandoned_mineshaft.json:261]: yeah... why would you do this?

**Comment by inxomnyaa** [comment]
I deleted it because its not worth to get legal issues because someone here \*cough you\* doesn't understand what i am aiming at

**Comment by AvgZing** [comment]
Legal issues are the worst and they are always tough to resolve (hint, hint? Functions?)

**Comment by inxomnyaa** [comment]
My idea simply was to let the chests not generate anything inside, except when the user manually places an addon into a folder

**Comment by HimbeersaftLP** [comment]
What about asking Mojang vie Email if they allow you to use it :D

**Comment by Ad5001** [comment]
@thebigsmileXD  You means it only generates when the users hits it? Then, just use the "processingLoots.json" file to gen id in async instead of using a none creatable tile ;).
@HimbeersaftLP That's a good idea !

**Comment by inxomnyaa** [comment]
@HimbeersaftLP they don't. I did ask.

**Comment by inxomnyaa** [comment]
> Btw why did you deleted this branch?

Because i hate it when i see someone that doesn't understand the code i made or what i say destroying my codechanges.

**Pull request closed without merging**